### PR TITLE
Collect metrics from GCS data

### PR DESCRIPTION
**Pull request by mdittmer**
Main flow:

- Look up latest (4) runs
- Download >100k GCS objects containing run test results in parallel
- Rehash downloaded data into `{testId: {platformId: [results]}}`
- For each test directory/file
  - Compute total number of tests+subtests under directory/file
  - Bucket those tests by "0 platforms passing", "1 platform passing", ..., "n platforms passing"
- For each browser name
  - Compute lists of test names for "this browser and 0 others failing", "this browser and 1 other failing", ..., "this browser and n others failing"
- Push all computed results to a GCS bucket

Other notes:
- Logging goes to a file
- Console output updates report on GCS download (since that is the longest-wait-time-highest-risk operation), then nothing until completion
- Default output bucket is new `wptd-metrics` bucket organized by `<unix timestamp for download start>-<unix timestamp for download end>` directory for each run

Future ideas (plz comment with thoughts):
- Inputs and/or outputs from/to:
  - ((nowhere)) (output only; dry run)
  - file
  - ???

//cc @jeffcarp as this is all golang

### CODE REVIEW

**Comment by mdittmer** [comment]
CI probably failing because this depends on #313

File: metrics/compute/compute.go [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+type TestRunsStatus map[metrics.TestId]map[base.TestRun]metrics.CompleteTestStatus
+
+// Type for decision problem: "What does it mean for a test result to 'pass'"?
+type Passes func(*metrics.CompleteTestStatus) bool
+
+//
+// Passes functions
+//
+
+func OkAndUnknonwOrPasses(status *metrics.CompleteTestStatus) bool {
```
Comment (rwaldron) [inline metrics/compute/compute.go]: Typo: `OkAndUnknonwOrPasses` => `OkAndUnknownOrPasses`

If you want to match the naming: `OkAndUnknownOrPass`

**Comment by mdittmer** [comment]
*ping*

**Comment by mdittmer** [comment]
*ping*

**Comment by lukebjerring** [comment]
As per offline discussion about review flows, can you fast-forward your master branch from upstream, and rebase this branch.

(either of the following options is fine:)

    git reset master  && git commit -a -m "Collect metrics from GCS data" && git push -f

or

   git rebase -i master

which will of course allow you to scrutinize and squash or drop extraneous commits from the log as per your selection.

**Pull request merged**